lib/core/src/Cardano/Wallet/TokenMetadata.hs

```diff
@@ -450,8 +453,12 @@ instance FromJSON PropertyName where
     parseJSON = withText "PropertyName" (pure . PropertyName)
 
 instance FromJSON BatchResponse where
-    parseJSON =  withObject "BatchResponse" $ \o ->
-        BatchResponse <$> o .: "subjects"
+    parseJSON =  withObject "BatchResponse" $ \o -> do
+        (a :: Value) <- o .: "subjects"
+        let maybeParseItem v = fmap Just (parseJSON v) <|> pure Nothing
+        BatchResponse <$> withArray "items"
+            (fmap catMaybes . mapM maybeParseItem . toList)
+            a
```

Comment:
We could avoid the `toList` traversal here by parsing it as a list of Value in the first place:

```hs
parseJSON =  withObject "BatchResponse" $ \o -> do
    (xs :: [Value]) <- o .: "subjects"
    let maybeParseItem v = fmap Just (parseJSON v) <|> pure Nothing
    BatchResponse <$> (catMaybes <$> mapM maybeParseItem xs)
```